conda_forge_tick/migrators.xsh

```diff
@@ -140,43 +170,27 @@ class Version(Migrator):
 
     def migrate(self, recipe_dir, attrs, hash_type='sha256'):
         # Render with new version but nothing else
+        version = attrs['new_version']
         with indir(recipe_dir):
+            with open('meta.yaml', 'r') as f:
+                text = f.read()
+        url = re.search('  url:.*?\n(    -.*\n?)*', text).group()
+        if 'cran.r-project.org/src/contrib' in url:
+            version = version.replace('_', '-')
+        with indir(recipe_dir), ${...}.swap(VERSION=version):
             for f, p, n in self.patterns:
                 p = eval_version(p)
                 n = eval_version(n)
                 replace_in_file(p, n, f)
             with open('meta.yaml', 'r') as f:
                 text = f.read()
-            # If we can't parse the meta_yaml then jump out
-            meta_yaml = parsed_meta_yaml(text)
-            # If the parser returns None, then we didn't read the meta.yaml
-            # TODO: How we didn't fail at 01 on this recipe is mysterious
-            if meta_yaml is None:
-                attrs['bad'] = '{}: failed to read meta.yaml\n'.format($PROJECT)
-                return False
-            source_url = meta_yaml.get('source', {}).get('url')
-            if not source_url:
-                attrs['bad'] = '{}: missing url\n'.format($PROJECT)
-                return False
-            if isinstance(source_url, list):
-                for url in source_url:
-                    if 'Archive' not in url:
-                        source_url = url
-                        break
-            if 'cran.r-project.org/src/contrib' in source_url:
-                $VERSION = $VERSION.replace('_', '-')
-
-        # now, update the feedstock to the new version
-        source_url = eval_version(source_url)
-        try:
-            hash = hash_url(source_url, hash_type)
-        except urllib.error.HTTPError:
-            attrs['bad'] = '{}: hash failed at {}\n'.format(
-                meta_yaml.get('package', {}).get('name', 'UNKOWN'), source_url)
-            return False
-
-        new_patterns = tuple(self.patterns) + tuple(self.more_patterns)
-        with indir(recipe_dir), ${...}.swap(HASH_TYPE=hash_type, HASH=hash, SOURCE_URL=source_url):
+
+        # Get patterns to replace checksum for each platform
+        rendered_text = render_meta_yaml(text)
+        urls = self.find_urls(rendered_text)
+        new_patterns = self.get_hash_patterns('meta.yaml', urls, hash_type)
+
+        with indir(recipe_dir):
             for f, p, n in new_patterns:
```

Comment:
Doesn't this beer the env swap too?

Reply:
None of the patterns in `new_patterns` use environment variables.

Reply:
Ok cool, thanks!